packages/data-context/test/unit/codegen/spec-options.spec.ts

```diff
@@ -242,6 +242,35 @@ describe('spec-options', () => {
         expect(result.codeGenType).to.eq('e2e')
         expect(result.fileName).to.eq(`TestName.foo.bar-copy-2.js`)
       })
+
+      context('file name contains special characters', async () => {
+        [
+          { condition: 'braces', fileName: '[...MyComponent].vue', expectedFileName: '[...MyComponent].cy.js', expectedComponentName: 'MyComponent' },
+          { condition: 'hyphens', fileName: 'my-component.vue', expectedFileName: 'my-component.cy.js', expectedComponentName: 'mycomponent' },
+          { condition: 'parentheses', fileName: 'My(Component).js', expectedFileName: 'My(Component).cy.js', expectedComponentName: 'MyComponent' },
+          { condition: 'period-separated', fileName: 'my.component.js', expectedFileName: 'my.component.cy.js', expectedComponentName: 'my' },
```

Comment:
Angular has the `my.component.ts` convention, I'd expect the output to be `MyComponent` rather than 	my`. Same for the `my-component.vue` where I'd expect it to be pascal case.

Reply:
@ZachJW34 This may just be my React background talking - I would want a generated spec for `ContactUs.component.tsx` to result in code like `<ContactUs />` rather than `<ContactUsComponent />`. Makes more sense in a JSX context (to me at least), but I can definitely see where it would make more sense to tack on the `...Component` for Angular and other non-JSX situations.

We don't support this flow for Angular yet so this is technically a non-issue and could be tackled when we get to implementing it, but I hesitate to do too much language-specific logic in here since it would become a mess pretty quickly. I could see just going with including the `Component` suffix and calling it good since it would be a simple rename for the React devs out there who are opinionated like me

Reply:
I'll also look into PascalCase for period- and hyphen-delimited names